modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -3241,20 +3241,94 @@ void ONNXImporter::parseQuantDequant(LayerParams& layerParams, const opencv_onnx
+        if(zpMat.total() > 1) scalarParams = false;
+    }
+
+    if (constBlobs.find(node_proto.input(0)) == constBlobs.end()) // Variable input.
+    {
+        if (!scalarParams) // 1-D tensor quantized params.
+            CV_Error(Error::StsNotImplemented, "When scale or zero point is a 1-D tensor, "
+                                               "the quantization layer does not support variable input.");
 
-    layerParams.set("scales", scale);
-    layerParams.set("zeropoints", zeropoint);
+        // NOTE: For Dequantize, the zeropoint could be int32 or int8. For other layer, zeropoint can only be int8. (ai.onnx opset 15).
+        int zeropoint = zpMat.empty() ? 0 : zpMat.depth() == CV_32S ?
+                getScalarFromMat<int>(zpMat) : (int)getScalarFromMat<int8_t>(zpMat);
+        float scale = getScalarFromMat<float>(scaleMat);
+        layerParams.set("scales", scale);
+        layerParams.set("zeropoints", zeropoint);
 
-    if (layerParams.type == "Quantize")
-        layerParams.set("depth", CV_8S);
-    else // Dequantize
-        layerParams.set("depth", CV_32F);
+        if (layerParams.type == "Quantize")
+            layerParams.set("depth", CV_8S);
+        else // Dequantize
+            layerParams.set("depth", CV_32F);
 
-    addLayer(layerParams, node_proto);
+        addLayer(layerParams, node_proto);
+    }
+    else
+    {
+        Mat inputBuf = getBlob(node_proto, 0);
+        Mat output;
+
+        Mat zpMat_double = zpMat;
+        if (!zpMat.empty())
+            zpMat.convertTo(zpMat_double, CV_64F);
+        Mat scaleMat_double;
+        scaleMat.convertTo(scaleMat_double, CV_64F);
+
+        if (layerParams.type == "Quantize")
+        {
+            if (scalarParams)
+            {
+                float scale = getScalarFromMat<float>(scaleMat);
+                int zeropoint = zpMat.empty() ? 0 : (int)getScalarFromMat<int8_t>(zpMat);
+                inputBuf.convertTo(output, CV_8S,
+                                   1.f/ scale, zeropoint);
+            }
+            else
+            {
+                if (zpMat_double.empty())
+                    output = inputBuf;
+                else
+                    subtract(inputBuf, zpMat_double.row(0), output);
+
+                multiply(output, scaleMat, output);
+                output.convertTo(output, CV_8S);
+            }
+        }
+        else // "Dequantize"
+        {
+            if (scalarParams)
+            {
+                float scale = getScalarFromMat<float>(scaleMat);
+                int zeropoint = zpMat.empty() ? 0 : zpMat.depth() == CV_32S ?
+                        getScalarFromMat<int>(zpMat) : (int)getScalarFromMat<int8_t>(zpMat);
+                inputBuf.convertTo(output, CV_32F, scale, -(scale*zeropoint));
+            }
+            else
+            {
+                inputBuf.convertTo(output, CV_32F);
+                if (!zpMat_double.empty())
+                    subtract(output, zpMat_double.row(0), output);
```

Comment:
This code is no good. 
For `Quantized` and `Dequantized` layer, the zeropoint and scale can be 1-D tensor or Scalar. And if it is 1-D tensor, the `subtract` and `multiply` will have problems. For example, output shape is `[5, 3, 3, 3]` and the scale or zeropoint shape is `[5, 1]`. The `subtract` can only works when the scale or zeropoint is `CV_64F`. [details here](https://github.com/opencv/opencv/blob/4.x/modules/core/src/arithm.cpp#L652). 

Any suggestions to support this kind of calculation? like `[5, 3, 3, 3]` multiply `[5, 1]`. For now, as a workaround, I convert the second Mat to Scalar.

Reply:
we could do something like:

```cpp
vector<Range> idx(input.dims, Range::all());
for (size_t i = 0; i < zeropoint.size(); ++i)
{
    idx[axis] = Range(i, i+1);
    ((input(idx) - zeropoint[i])*scale[i]).copyTo(output(idx));
}
```

not sure, how to do this better. @fengyuentau, what do you think?

Reply:
Not sure what `axis` is but we can manually broadcast the operand like in `parseExpand`.
https://github.com/opencv/opencv/blob/e80b443cd93012f1fdd79baf7ed325beff606a89/modules/dnn/src/onnx/onnx_importer.cpp#L2463-L2471

Reply:
https://github.com/onnx/onnx/blob/main/docs/Operators.md#dequantizelinear
https://github.com/onnx/onnx/blob/main/docs/Operators.md#quantizelinear

axis is used for per axis quantization.